pkg/apis/core/validation/validation.go

```diff
@@ -3682,6 +3701,57 @@ func validateHostUsers(spec *core.PodSpec, fldPath *field.Path) field.ErrorList
 	return allErrs
 }
 
+// validateFileKeyRefVolume validates that volumes referenced by FileKeyRef environment variables
+// are of type emptyDir. FileKeyRef requires emptyDir volumes to ensure proper file access.
+func validateFileKeyRefVolume(spec *core.PodSpec, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	// Collect all volume names referenced by FileKeyRef environment variables
+	referencedVolumeNames := sets.Set[string]{}
+	for _, container := range spec.Containers {
+		for _, env := range container.Env {
+			if env.ValueFrom != nil && env.ValueFrom.FileKeyRef != nil {
+				referencedVolumeNames.Insert(env.ValueFrom.FileKeyRef.VolumeName)
+			}
+		}
+	}
+
+	existingVolumeNames := sets.Set[string]{}
+	for _, volume := range spec.Volumes {
+		existingVolumeNames.Insert(volume.Name)
+	}
+
+	// Check if all referenced volumes exist
+	for i, container := range spec.Containers {
+		for j, env := range container.Env {
+			if env.ValueFrom != nil && env.ValueFrom.FileKeyRef != nil {
+				volumeName := env.ValueFrom.FileKeyRef.VolumeName
+				if !existingVolumeNames.Has(volumeName) {
+					allErrs = append(allErrs, field.NotFound(
+						fldPath.Child("containers").Index(i).Child("env").Index(j).Child("valueFrom").Child("fileKeyRef").Child("volumeName"),
+						fmt.Sprintf("volume %q referenced by FileKeyRef does not exist", volumeName),
```

Comment:
The value argument is not meant to be a printf, and the field name is already part of the error.  Just pass `volumeName`.